### ISSUE TITLE
cannot link with https://github.com/Tarendai/Shard

### ISSUE THREAD

**Issue by eronoobos** [eronoobos/ShardSpringLua#3]
Currently this repo is independent from https://github.com/Tarendai/Shard, which makes things difficult. It would be better if this repo used https://github.com/Tarendai/Shard as a submodule somehow, but only the parts it needs.

**Comment by tomjn** [comment]
I'm willing to change the folder structure of the main repo if needed, be it separating out the lua part, or separating out the C++ part into a new repo, what do you think?

**Comment by eronoobos** [comment]
Yes, the C++ part and the Lua part would have to be different repos. Everything in data/ai/ would be in the Lua repo. It would then be included in the C++ repo and ShardSpringLua repo as a submodule. That would be a relief. More complicated in a way, but at least I wouldn't be using local hooks to mirror my commits to ShardSpringLua to Shard, and there would be one version control on the lua code.

**Comment by tomjn** [comment]
I'm unsure what problems may occur with turning it into a submodule of the C++ AI, since it already contains submodules, perhaps @abma can advise?

**Comment by eronoobos** [comment]
in my own attempts, even just updating the Shard submodule within Spring, with the BA and BAR submodules intact is a pain. For reasons I can't understand, the EvoRTS submodule is fetched in the first recursion when I clone Spring, but not BA or BAR. Whatever EvoRTS is doing, that's the right way apparently.

**Comment by tomjn** [comment]
It's possible that it's a special case rather than some dynamic system in
the spring build system. Abma will know more

**Comment by eronoobos** [comment]
Hm, maybe. I think the problem is that EvoRTS was already a submodule in the version of Shard that's currently in the Spring repo, whereas BA and BAR were not. I think I'm just not sure how to use git to handle this case.

**Comment by tomjn** [comment]
BA and BAR are also submodules in https://github.com/Tarendai/Shard/tree/master/data/ai so that's not it, I still think EvoRTS is a special handcoded case, but I don't know where to look to know

**Comment by eronoobos** [comment]
right, but Spring hasn't updated its Shard submodule since BA and BAR became submodules within it

**Comment by tomjn** [comment]
For reference, there's a ticket here https://github.com/Tarendai/Shard/issues/102
